### PR TITLE
[#1202] improvement: Add HealthScriptChecker for execute special health check shell script

### PR DESCRIPTION
**Pull request by xumanbu**
### What changes were proposed in this pull request?

add HealthScriptChecker for execute special health check shell script
ref: https://github.com/apache/hadoop/blob/trunk/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/health/NodeHealthScriptRunner.java

### Why are the changes needed?

Fix:  #1202

### Does this PR introduce _any_ user-facing change?
`rss.server.health.checker.script.path` : The health script file path for HealthScriptChecker, if script file should have execute permission.

`rss.server.health.checker.script.exe.timeout`: The health script file execute timeout seconds

### How was this patch tested?
add ut

### CODE REVIEW

**Comment by jerqi** [comment]
Could you add some documents for this pr?

File: server/src/test/java/org/apache/uniffle/server/HealthScriptCheckerTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  private String getScriptFilePath(String fileName) {
+    return Objects.requireNonNull(this.getClass().getClassLoader().getResource(fileName)).getFile();
+  }
+
+  private boolean setExecutable(String filePath) {
+    File file = new File(filePath);
+    file.setExecutable(true);
+    return file.canExecute();
+  }
+
```
Comment (jerqi) [inline server/src/test/java/org/apache/uniffle/server/HealthScriptCheckerTest.java:62]: Could you remove extra blank line?

File: server/src/test/resources/healthy-script1.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+echo "NORMAL: Check Disk is Normal"
```
Comment (jerqi) [inline server/src/test/resources/healthy-script1.sh:2]: Could you add license header for this file?

File: server/src/test/resources/healthy-script2.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
```
Comment (jerqi) [inline server/src/test/resources/healthy-script2.sh:1]: Could you add license header for this file?

File: server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java [hunk 4]
```diff
@@ -271,6 +271,20 @@ public class ShuffleServerConf extends RssBaseConf {
...
 
+  public static final ConfigOption<String> HEALTH_CHECKER_SCRIPT_PATH =
+      ConfigOptions.key("rss.server.health.checker.script.path")
+          .stringType()
+          .defaultValue("")
+          .withDescription(
+              "The health script file path for HealthScriptChecker, if script file should have execute permission.");
+
+  public static final ConfigOption<Long> HEALTH_CHECKER_SCRIPT_EXE_TIMEOUT =
+      ConfigOptions.key("rss.server.health.checker.script.exe.timeout")
```
Comment (zuston) [inline server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java:282]: `exe` -> `execution` ?

File: server/src/main/java/org/apache/uniffle/server/HealthScriptChecker.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    this.scriptTimeout = conf.getLong(ShuffleServerConf.HEALTH_CHECKER_SCRIPT_EXE_TIMEOUT);
+  }
+
+  @Override
+  public boolean checkIsHealthy() {
+    HealthCheckerExitStatus status = HealthCheckerExitStatus.SUCCESS;
+    // always build executor is to load the latest script, the script file can update in need.
+    Shell.ShellCommandExecutor commandExecutor = new Shell.ShellCommandExecutor(new String[] {healthScriptPath}, null, null, scriptTimeout);
+    try {
+      commandExecutor.execute();
```
Comment (zuston) [inline server/src/main/java/org/apache/uniffle/server/HealthScriptChecker.java:50]: So the ExitCodeException will be thrown when the execution failed?

Comment (xumanbu) [reply server/src/main/java/org/apache/uniffle/server/HealthScriptChecker.java:50]: if shell execute stopped exit 1, throw ExitCodeException. I'll add a test case.

File: server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java (hunk 4 above)
Comment (xumanbu) [reply server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java:282]: ok

File: server/src/main/java/org/apache/uniffle/server/HealthScriptChecker.java (hunk 5 above)
Comment (zuston) [reply server/src/main/java/org/apache/uniffle/server/HealthScriptChecker.java:50]: Got it.

**Comment by xumanbu** [comment]
> Could you add some documents for this pr?

I'll done

**Comment by jerqi** [comment]
> > Could you add some documents for this pr?
> 
> I'll done

Could you add the document in the https://github.com/apache/incubator-uniffle/blob/master/docs/server_guide.md ?

**Review by jerqi — APPROVED** [review]
LGTM. thanks @zuston @xumanbu

**Review by zuston — APPROVED** [review]

**Pull request merged**